Финальное задание 8 спринта

karabasBaRaBaS-1276 opened the pull request.

Commit:
Реализация основного функционала по работе с хранилищем

Commit:
Изменение подключения к БД + первый тест

Commit:
Дописал тесты. Поправил ошибки, найденные при написании тестов

ivannizh commented on parcel.go:
```diff
@@ -2,59 +2,251 @@ package main
+		sql.Named("client", client),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("ошибка при поиске списка посылок по клиенту: %w", err)
```
Очень хорошо описываешь ошибки, тебе поможет это в будущем, но лучше на английском это делать

ivannizh commented on parcel.go:
```diff
@@ -2,59 +2,251 @@ package main
+	if status == "" {
+		return fmt.Errorf("ошибка при изменении статуса: статус не может быть пустым")
+	}
+	if number == 0 {
```
```suggestion
	if number <= 0 {
```

ivannizh commented on parcel_test.go:
```diff
@@ -28,60 +30,143 @@ func getTestParcel() Parcel {
-	// проверьте, что значения всех полей в полученном объекте совпадают со значениями полей в переменной parcel
+	p, err := store.Get(number)
+	assert.NoError(t, err, "Нет ошибок при получении информации о посылке")
+	assert.Equal(t, testParcel.Client, p.Client, "Значение клиента совпадает")
```
Функция `Equal` может сравнивать структуры сразу (а не по полям), надо только с полем `Number` разобраться.

Нужно переделать, так как при таком подходе при изменении структуры нужно будет менять все тесты

ivannizh replied:
Только это нужно исправить для новой итерации!

ivannizh reviewed (commented):
Привет! Хорошая работа! Осталось исправить некоторые моменты и работа будет принята.

Commit:
Мелкие изменения:
1. Тексты ошибок по-английски
2. Условие проверки number == 0 заменено на number <= 0
3. Тест упрощен тест TestAddGetDelete: вместо Equal каждого поля структуры сделан EqualValues всех полей структуры. Тем самым тест нужно переписывать при расширении структуры.

ivannizh reviewed (approved):
Отлично, принято!
